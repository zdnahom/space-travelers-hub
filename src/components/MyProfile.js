import { useSelector } from 'react-redux';

const MyProfile = () => {
  const rockets = useSelector((state) => state.rockets.rockets);
  const reservedRockets = rockets.filter((rocket) => rocket.reserved);
  const { missions } = useSelector((store) => store.missions);
  const joinedMissions = missions.filter((mission) => mission.reserved === true);

  return (
    <section>
     <div>
        <h2 className="profile-headline-text">My Missions</h2>
        {joinedMissions.length > 0 ? (
          <ul className="reserved-items-container">
            {joinedMissions.map((mission) => (
              <li key={mission.mission_id} className="reserved-item">
                <span className="reserved-item-name">{mission.mission_name}</span>
              </li>
            ))}
          </ul>
        ) : (
          <div className="no-reserved">You haven&apos;t joined any missions yet.</div>
        )}
      </div>
      <div>
        <h2 className="profile-headline-text">My Rockets</h2>
        {reservedRockets.length > 0 ? (
          <ul className="reserved-items-container">
            {reservedRockets.map((rocket) => (
              <li key={rocket.id} className="reserved-item">
                <span className="reserved-item-name">{rocket.name}</span>
              </li>
            ))}
          </ul>
        ) : (
          <div className="no-reserved">You haven&apos;t reserved any rockets yet.</div>
        )}
      </div>
    </section>
  );
};

export default MyProfile;
