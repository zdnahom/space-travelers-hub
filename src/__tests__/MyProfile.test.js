import { render, cleanup } from '@testing-library/react';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import MyProfile from '../components/MyProfile';

afterEach(cleanup);
const mockStore = configureMockStore();
const store = mockStore({
  rockets: {
    rockets: [
      {
        id: '1',
        image: 'https://i.imgur.com/DaCfMsj.jpg',
        name: 'Falcon 1',
        description: 'The first rocket',
        reserved: false,
      },
      {
        id: '2',
        image: 'https://farm1.staticflickr.com/929/28787338307_3453a11a77_b.jpg',
        name: 'Falcon 9',
        description: 'The second rocket',
        reserved: false,
      },
    ],
  },
  missions: {
    missions: [
      {
        mission_id: '1',
        mission_name: 'Thaicom',
        description: 'The first mission',
        reserved: false,
      },
      {
        mission_id: '2',
        mission_name: 'Telstar',
        description: 'The second mission',
        reserved: false,
      },
    ],
    isFetched: false,
  },
});
describe('MyProfile component test', () => {
  it('It renders', () => {
    const { container } = render(
      <Provider store={store}>
        <MyProfile />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
