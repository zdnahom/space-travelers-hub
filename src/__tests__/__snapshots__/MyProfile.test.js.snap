// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProfile component test It renders 1`] = `
<div>
  <section
    class="profile"
  >
    <div>
      <h2
        class="profile-headline-text"
      >
        My Missions
      </h2>
      <div
        class="no-reserved"
      >
        You haven't joined any missions yet.
      </div>
    </div>
    <div>
      <h2
        class="profile-headline-text"
      >
        My Rockets
      </h2>
      <div
        class="no-reserved"
      >
        You haven't reserved any rockets yet.
      </div>
    </div>
  </section>
</div>
`;
